### ISSUE TITLE
deeplearning4j-nlp 0.4 generates error not seen in 0.0.3.3.4.alpha2

### ISSUE THREAD

**Issue by ozborn** [deeplearning4j/deeplearning4j#743]
When creating a simple Word2Vec Model like this:
```
		Word2Vec vec = new Word2Vec.Builder().sampling(1e-5)
				.minWordFrequency(5).batchSize(1000).useAdaGrad(false).layerSize(layerSize)
				.iterations(3).learningRate(0.025).minLearningRate(1e-2).negativeSample(10)
				.iterate(iter).tokenizerFactory(t).build();
		try {
			vec.fit();
		} catch (IOException e) {
			e.printStackTrace();
		}
```
and using [a non-trivial amount of text] (http://biotext.berkeley.edu/data/dis_treat_data/labeled_abstracts.txt) I can generate the stack trace shown at the end of this post due to this error: 
```
java.lang.NoSuchMethodError: org.nd4j.linalg.factory.BlasWrapper.level1()Lorg/nd4j/linalg/api/blas/Level1;
```

I can workaround this problem in 2 ways:
1)  Use only a few lines of text
2) Revert back to  deeplearning4j-nlp 0.0.3.3.4.alpha2.

The error appears to be introduced in 0.4rc0 since that also fails giving the same error as the current maven version 0.4-rc3.3.

Has this methodology for building models been updated in 0.4? I can submit a Junit4 test code if needed. Stack trace below, any help appreciated.
```
java.lang.NoSuchMethodError: org.nd4j.linalg.factory.BlasWrapper.level1()Lorg/nd4j/linalg/api/blas/Level1;
	at org.deeplearning4j.models.embeddings.inmemory.InMemoryLookupTable.iterateSample(InMemoryLookupTable.java:227)
	at org.deeplearning4j.models.word2vec.Word2Vec.iterate(Word2Vec.java:343)
	at org.deeplearning4j.models.word2vec.Word2Vec.skipGram(Word2Vec.java:331)
	at org.deeplearning4j.models.word2vec.Word2Vec.trainSentence(Word2Vec.java:308)
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:210)
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:197)
	at org.deeplearning4j.parallel.Parallelization$3.call(Parallelization.java:160)
	at akka.dispatch.Futures$$anonfun$future$1.apply(Future.scala:94)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Uncaught error from thread [default-akka.actor.default-dispatcher-22] shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[default]
java.lang.NoSuchMethodError: org.nd4j.linalg.factory.BlasWrapper.level1()Lorg/nd4j/linalg/api/blas/Level1;
	at org.deeplearning4j.models.embeddings.inmemory.InMemoryLookupTable.iterateSample(InMemoryLookupTable.java:227)
	at org.deeplearning4j.models.word2vec.Word2Vec.iterate(Word2Vec.java:343)
	at org.deeplearning4j.models.word2vec.Word2Vec.skipGram(Word2Vec.java:331)
	at org.deeplearning4j.models.word2vec.Word2Vec.trainSentence(Word2Vec.java:308)
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:210)
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:197)
	at org.deeplearning4j.parallel.Parallelization$3.call(Parallelization.java:160)
	at akka.dispatch.Futures$$anonfun$future$1.apply(Future.scala:94)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Uncaught error from thread [default-akka.actor.default-dispatcher-20] shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[default]
java.lang.NoSuchMethodError: org.nd4j.linalg.factory.BlasWrapper.level1()Lorg/nd4j/linalg/api/blas/Level1;
	at org.deeplearning4j.models.embeddings.inmemory.InMemoryLookupTable.iterateSample(InMemoryLookupTable.java:227)
	at org.deeplearning4j.models.word2vec.Word2Vec.iterate(Word2Vec.java:343)
	at org.deeplearning4j.models.word2vec.Word2Vec.skipGram(Word2Vec.java:331)
	at org.deeplearning4j.models.word2vec.Word2Vec.trainSentence(Word2Vec.java:308)
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:210)
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:197)
	at org.deeplearning4j.parallel.Parallelization$3.call(Parallelization.java:160)
	at akka.dispatch.Futures$$anonfun$future$1.apply(Future.scala:94)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Uncaught error from thread [
```

**Comment by agibsonccc** [comment]
Which nd4j backend are you using?

**Comment by ozborn** [comment]
nd4j-jblas

Your insight was correct  - they were out of sync for the 0.4 version (but not for the functional 0.0.3 version). That version is still working correctly, but the 0.4 version is unfortunately still giving me an error (albeit a different one) as shown below.

Any thoughts appreciated - I assume none of this is due to my lack of setting a lookup table or cache for the model?

```
5:09:41.431 [default-akka.actor.default-dispatcher-18] INFO  Cluster(akka://default) - Cluster Node [akka.tcp://default@0.0.0.0:39326] - No seed-nodes configured, manual cluster join required
15:09:41.449 [default-akka.actor.default-dispatcher-20] WARN  o.d.parallel.Parallelization - Error occurred processing data
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.nd4j.linalg.api.shape.Shape.getOffset(Shape.java:498) ~[nd4j-api-0.4-rc3.3.jar:na]
	at org.nd4j.linalg.api.shape.Shape.getDouble(Shape.java:249) ~[nd4j-api-0.4-rc3.3.jar:na]
	at org.nd4j.linalg.api.ndarray.BaseNDArray.getDouble(BaseNDArray.java:1440) ~[nd4j-api-0.4-rc3.3.jar:na]
	at org.nd4j.linalg.api.ndarray.BaseNDArray.getInt(BaseNDArray.java:1428) ~[nd4j-api-0.4-rc3.3.jar:na]
	at org.deeplearning4j.models.embeddings.inmemory.InMemoryLookupTable.iterateSample(InMemoryLookupTable.java:255) ~[deeplearning4j-nlp-0.4-rc3.3.jar:na]
	at org.deeplearning4j.models.word2vec.Word2Vec.iterate(Word2Vec.java:343) ~[deeplearning4j-nlp-0.4-rc3.3.jar:na]
	at org.deeplearning4j.models.word2vec.Word2Vec.skipGram(Word2Vec.java:331) ~[deeplearning4j-nlp-0.4-rc3.3.jar:na]
	at org.deeplearning4j.models.word2vec.Word2Vec.trainSentence(Word2Vec.java:308) ~[deeplearning4j-nlp-0.4-rc3.3.jar:na]
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:210) ~[deeplearning4j-nlp-0.4-rc3.3.jar:na]
	at org.deeplearning4j.models.word2vec.Word2Vec$3.run(Word2Vec.java:197) ~[deeplearning4j-nlp-0.4-rc3.3.jar:na]
	at org.deeplearning4j.parallel.Parallelization$3.call(Parallelization.java:160) ~[deeplearning4j-scaleout-akka-0.4-rc3.3.jar:na]
	at akka.dispatch.Futures$$anonfun$future$1.apply(Future.scala:94) ~[akka-actor_2.10-2.3.4-spark.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[scala-library-2.10.4.jar:na]
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41) [akka-actor_2.10-2.3.4-spark.jar:na]
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393) [akka-actor_2.10-2.3.4-spark.jar:na]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [scala-library-2.10.4.jar:na]
15:09:41.454 [default-akka.actor.default-dispatcher-20] WARN  o.d.parallel.Parallelization - Error occurred processing data
```

**Comment by agibsonccc** [comment]
Which version are you using?

**Comment by ozborn** [comment]
Function versions:
nd4j 0.0.3.5.5.5
dl4j 0.0.3.3.4.alpha2

Versions with error - both 0.4-rc3.3.

Have not tested with earlier versions of 0.4 since I put it in sync with nd4j

**Comment by agibsonccc** [comment]
Link your pom please,

**Comment by ozborn** [comment]
A functional version of the pom file (https://gist.github.com/ozborn/866cd888c875200ad1dc). Switching it to use 0.4-rc3.3 as in this file (https://gist.github.com/ozborn/9b066bb4cef6dbbec649) gives me the error.

**Comment by agibsonccc** [comment]
Try this version for nd4j: 	0.4-rc3.4

**Comment by agibsonccc** [comment]
Could you also try https://github.com/deeplearning4j/dl4j-0.4-examples/blob/master/src/main/java/org/deeplearning4j/examples/word2vec/Word2VecRawTextExample.java 

?

I haven't seen that and I'm not sure where it'd be coming from ;/. I'm wondering if it's your environment.

**Comment by ozborn** [comment]
No luck with nd4j: 0.4-rc3.4, will try example next.

**Comment by agibsonccc** [comment]
@ozborn anything on this? I"m able to run that fine out of the box.

**Comment by agibsonccc** [comment]
@ozborn so my unit tests here are passing: https://github.com/deeplearning4j/deeplearning4j/blob/1936b04cf9f5c9f691705d50fa25dcc34aea6c72/deeplearning4j-scaleout/deeplearning4j-nlp/src/test/java/org/deeplearning4j/models/word2vec/Word2VecTests.java#L85-84

I need some more information on your problem here.

**Comment by agibsonccc** [comment]
Closing due to lack of reproducibility. Just wait till I get the release and try again with the right versions. Happy to talk in gitter about helping you build that model. It's only a few lines of code.

**Comment by ozborn** [comment]
@agibsonccc  - thanks for all your help.
 I am still trying to pin this down, let me tell you what I know:
I can run your Word2VecRawTextExample as 
``` java -cp ./:target/classes org.deeplearning4j.examples.word2vec.Word2VecRawTextExample```
and it works just fine.
I can also incorporate this into my project with my pom.xml file and it works.
I can substitute my own data for your raw_sentences and it works just fine.
I can also get my original problem to work if I use the Word2VecRawTextExample instead of my original model as shown below:
```
        InMemoryLookupCache cache = new InMemoryLookupCache();
        WeightLookupTable table = new InMemoryLookupTable.Builder()
                .vectorLength(100)
                .useAdaGrad(false)
                .cache(cache)
                .lr(0.025f).build();

        log.info("Building model....");
        Word2Vec vec = new Word2Vec.Builder()
                .minWordFrequency(5).iterations(1)
                .layerSize(100).lookupTable(table)
                .stopWords(new ArrayList<String>())
                .vocabCache(cache).seed(42)
                .windowSize(5).iterate(iter).tokenizerFactory(t).build();


                /*
                Word2Vec vec = new Word2Vec.Builder().sampling(1e-5)
                                .minWordFrequency(5).batchSize(1000).useAdaGrad(false).layerSize(layerSize)
                                .iterations(3).learningRate(0.025).minLearningRate(1e-2).negativeSample(10)
                                .iterate(iter).tokenizerFactory(t).build();
                */
```
I do not yet know why this is, but it works around the ArrayOutOfBounds issue. Any thoughts on this?

I also get a separate problem (not my original error) when I incorporate Word2VecRawTextExample as a JUnit4 original test but that is a separate issue.

**Comment by agibsonccc** [comment]
Ok now THIS is helpful. Could you get me a github project or a zip file with the code in it?

**Comment by ozborn** [comment]
Yes, but not until late tonight or tomorrow morning. :(

**Comment by agibsonccc** [comment]
I'm in no hurry ;).

**Comment by ozborn** [comment]
Ha, your response speed makes me feel slow! ;)
I forked the examples project (https://github.com/ozborn/dl4j-0.4-examples) and added a few test files and a small bash script that reproduces the error. The moral of the story appears to be is that you can't get away without instantiating a lookup table and cache?

**Comment by agibsonccc** [comment]
@ozborn Could you try with the latest release (use the dl4j-0.4-examples versions for nd4j/canova/dl4j) Thanks!

**Comment by ozborn** [comment]
I'm now trying with canova 0.0.0.11 and nd4j 0.4-rc3.5 (up from 0.0.0.7 and nd4j 0.4-rc3.2).

Which dl4j version did you want me to try? I tried (dl4j 0.4-rc3.4 and rc3.3) but both gave me a set of different errors - I'm guessing this is because it isn't in sync with nd4j or it is trying to use native code.

```
ozborn@cargocult:~/code/repo/dl4j-0.4-examples$ ./test_bug.bsh 
16:22:14,413 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
16:22:14,414 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
16:22:14,414 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [file:/home/ozborn/code/repo/dl4j-0.4-examples/target/classes/logback.xml]
16:22:14,414 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs multiple times on the classpath.
16:22:14,415 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [jar:file:/home/ozborn/code/repo/dl4j-0.4-examples/target/deeplearning4j-examples-0.4-rc0-SNAPSHOT.jar!/logback.xml]
16:22:14,415 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [file:/home/ozborn/code/repo/dl4j-0.4-examples/target/classes/logback.xml]
16:22:14,499 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
16:22:14,507 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
16:22:14,516 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
16:22:14,552 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
16:22:14,618 |-INFO in ch.qos.logback.core.FileAppender[FILE] - File property is set to [logs/application.log]
16:22:14,620 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
16:22:14,622 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
16:22:14,625 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
16:22:14,629 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.catalina.core] to DEBUG
16:22:14,629 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.springframework] to DEBUG
16:22:14,629 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.deeplearning4j] to INFO
16:22:14,629 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.nd4j] to INFO
16:22:14,630 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [opennlp.uima.util] to OFF
16:22:14,630 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.uima] to OFF
16:22:14,630 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.cleartk] to OFF
16:22:14,630 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to ERROR
16:22:14,630 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
16:22:14,631 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
16:22:14,631 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
16:22:14,634 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@64b8f8f4 - Registering current configuration as safe fallback point

o.d.e.w.Word2VecRawTextExample - Load & Vectorize Sentences....
o.s.v.DataBinder - DataBinder requires binding of required fields [sentenceModelPath,sentenceTypeName]
o.s.v.DataBinder - DataBinder requires binding of required fields [rootFile]
Exception in thread "main" java.lang.ExceptionInInitializerError: unable to load from [netlib-native_system-linux-x86_64.so]
	at com.github.fommil.jni.JniLoader.load(JniLoader.java:81)
	at com.github.fommil.netlib.NativeSystemBLAS.<clinit>(NativeSystemBLAS.java:42)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.nd4j.linalg.cpu.BlasWrapper.<clinit>(BlasWrapper.java:41)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.nd4j.linalg.factory.Nd4j.initWithBackend(Nd4j.java:4532)
	at org.nd4j.linalg.factory.Nd4j.initContext(Nd4j.java:4490)
	at org.nd4j.linalg.factory.Nd4j.<clinit>(Nd4j.java:136)
	at org.deeplearning4j.models.embeddings.inmemory.InMemoryLookupTable$Builder.<init>(InMemoryLookupTable.java:564)
	at org.deeplearning4j.models.word2vec.Word2Vec$Builder.build(Word2Vec.java:634)
	at org.deeplearning4j.examples.word2vec.Word2VecRawTextExample2.main(Word2VecRawTextExample2.java:41)
```

**Comment by agibsonccc** [comment]
The new release forces the native system dependency. You need to link
libblas.so.3 and liblapack.so.3 for openblas with

ln -s /opt/OpenBLAS/lib/libopenblas.so /usr/lib64/libblas.so.3

ln -s /opt/OpenBLAS/lib/liblapack.so.3 /usr/lib64/liblapack.so.3

This is created after you do a make and make install on openblas

You can use pure java (slow) as well with the jvm argument
-Dorg.nd4j.linalg.cpu.force_native=false

I'm now trying with canova 0.0.0.11 and nd4j 0.4-rc3.5 (up from 0.0.0.7 and
nd4j 0.4-rc3.2).

Which dl4j version did you want me to try? I tried (dl4j 0.4-rc3.4 and
rc3.3) but both gave me a set of different errors - I'm guessing this is
because it isn't in sync with nd4j or it is trying to use native code.

ozborn@cargocult:~/code/repo/dl4j-0.4-examples$ ./test_bug.bsh
16:22:14,413 |-INFO in ch.qos.logback.classic.LoggerContext[default] -
Could NOT find resource [logback.groovy]
16:22:14,414 |-INFO in ch.qos.logback.classic.LoggerContext[default] -
Could NOT find resource [logback-test.xml]
16:22:14,414 |-INFO in ch.qos.logback.classic.LoggerContext[default] -
Found resource [logback.xml] at
[file:/home/ozborn/code/repo/dl4j-0.4-examples/target/classes/logback.xml]
16:22:14,414 |-WARN in ch.qos.logback.classic.LoggerContext[default] -
Resource [logback.xml] occurs multiple times on the classpath.
16:22:14,415 |-WARN in ch.qos.logback.classic.LoggerContext[default] -
Resource [logback.xml] occurs at
[jar:file:/home/ozborn/code/repo/dl4j-0.4-examples/target/deeplearning4j-examples-0.4-rc0-SNAPSHOT.jar!/logback.xml]
16:22:14,415 |-WARN in ch.qos.logback.classic.LoggerContext[default] -
Resource [logback.xml] occurs at
[file:/home/ozborn/code/repo/dl4j-0.4-examples/target/classes/logback.xml]
16:22:14,499 |-INFO in
ch.qos.logback.classic.joran.action.ConfigurationAction - debug
attribute not set
16:22:14,507 |-INFO in ch.qos.logback.core.joran.action.AppenderAction
- About to instantiate appender of type
[ch.qos.logback.core.FileAppender]
16:22:14,516 |-INFO in ch.qos.logback.core.joran.action.AppenderAction
- Naming appender as [FILE]
16:22:14,552 |-INFO in
ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming
default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for
[encoder] property
16:22:14,618 |-INFO in ch.qos.logback.core.FileAppender[FILE] - File
property is set to [logs/application.log]
16:22:14,620 |-INFO in ch.qos.logback.core.joran.action.AppenderAction
- About to instantiate appender of type
[ch.qos.logback.core.ConsoleAppender]
16:22:14,622 |-INFO in ch.qos.logback.core.joran.action.AppenderAction
- Naming appender as [STDOUT]
16:22:14,625 |-INFO in
ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming
default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for
[encoder] property
16:22:14,629 |-INFO in
ch.qos.logback.classic.joran.action.LoggerAction - Setting level of
logger [org.apache.catalina.core] to DEBUG
16:22:14,629 |-INFO in
ch.qos.logback.classic.joran.action.LoggerAction - Setting level of
logger [org.springframework] to DEBUG
16:22:14,629 |-INFO in
ch.qos.logback.classic.joran.action.LoggerAction - Setting level of
logger [org.deeplearning4j] to INFO
16:22:14,629 |-INFO in
ch.qos.logback.classic.joran.action.LoggerAction - Setting level of
logger [org.nd4j] to INFO
16:22:14,630 |-INFO in
ch.qos.logback.classic.joran.action.LoggerAction - Setting level of
logger [opennlp.uima.util] to OFF
16:22:14,630 |-INFO in
ch.qos.logback.classic.joran.action.LoggerAction - Setting level of
logger [org.apache.uima] to OFF
16:22:14,630 |-INFO in
ch.qos.logback.classic.joran.action.LoggerAction - Setting level of
logger [org.cleartk] to OFF
16:22:14,630 |-INFO in
ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level
of ROOT logger to ERROR
16:22:14,630 |-INFO in
ch.qos.logback.core.joran.action.AppenderRefAction - Attaching
appender named [STDOUT] to Logger[ROOT]
16:22:14,631 |-INFO in
ch.qos.logback.core.joran.action.AppenderRefAction - Attaching
appender named [FILE] to Logger[ROOT]
16:22:14,631 |-INFO in
ch.qos.logback.classic.joran.action.ConfigurationAction - End of
configuration.
16:22:14,634 |-INFO in
ch.qos.logback.classic.joran.JoranConfigurator@64b8f8f4 - Registering
current configuration as safe fallback point

o.d.e.w.Word2VecRawTextExample - Load & Vectorize Sentences....
o.s.v.DataBinder - DataBinder requires binding of required fields
[sentenceModelPath,sentenceTypeName]
o.s.v.DataBinder - DataBinder requires binding of required fields [rootFile]
Exception in thread "main" java.lang.ExceptionInInitializerError:
unable to load from [netlib-native_system-linux-x86_64.so]
    at com.github.fommil.jni.JniLoader.load(JniLoader.java:81)
    at com.github.fommil.netlib.NativeSystemBLAS.<clinit>(NativeSystemBLAS.java:42)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:264)
    at org.nd4j.linalg.cpu.BlasWrapper.<clinit>(BlasWrapper.java:41)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:264)
    at org.nd4j.linalg.factory.Nd4j.initWithBackend(Nd4j.java:4532)
    at org.nd4j.linalg.factory.Nd4j.initContext(Nd4j.java:4490)
    at org.nd4j.linalg.factory.Nd4j.<clinit>(Nd4j.java:136)
    at org.deeplearning4j.models.embeddings.inmemory.InMemoryLookupTable$Builder.<init>(InMemoryLookupTable.java:564)
    at org.deeplearning4j.models.word2vec.Word2Vec$Builder.build(Word2Vec.java:634)
    at org.deeplearning4j.examples.word2vec.Word2VecRawTextExample2.main(Word2VecRawTextExample2.java:41)

—
Reply to this email directly or view it on GitHub
<https://github.com/deeplearning4j/deeplearning4j/issues/743#issuecomment-144551642>
.

**Comment by ozborn** [comment]
The bug remains unfortunately, either using jblas or native even after merging with your latest updates in dl4j-0.4-examples. I used the pom file below for dependencies and I still see the bug, unless I create the model as in Word2VecRawTextExample5.

```
        <nd4j.version>0.4-rc3.5</nd4j.version>
        <dl4j.version>0.4-rc3.4</dl4j.version>
        <canova.version>0.0.0.11</canova.version>
        <jackson.version>2.5.1</jackson.version>
```
The test bug code was updated and pushed to (https://github.com/ozborn/dl4j-0.4-examples)

**Comment by agibsonccc** [comment]
@ozborn This is what I see running your version:

seTextVectorizer - Invoking finish on index
o.d.m.w.Word2Vec - Building binary tree
o.d.m.w.Word2Vec - Constructing priority queue
o.d.m.w.Word2Vec - Built tree
o.d.m.w.Word2Vec - Resetting weights
o.d.m.w.Word2Vec - Training word2vec multithreaded
o.s.v.DataBinder - DataBinder requires binding of required fields [rootFile]
o.d.m.w.Word2Vec - Processing sentences...
o.d.m.w.Word2Vec - Words so far 3988 with alpha at 0.024939732890367057
o.d.m.w.Word2Vec - Words so far 14481 with alpha at 0.024781161480793713
o.d.m.w.Word2Vec - Words so far 28755 with alpha at 0.02456545116913357
o.d.m.w.Word2Vec - Words so far 77103 with alpha at 0.023834810693573485
o.d.m.w.Word2Vec - Words so far 139104 with alpha at 0.022897844528991684


There's no bug here. If you can't get the binary deps to run. Then 
do -Dorg.nd4j.linalg.cpu.force_nativeorg.nd4j.linalg.cpu.force_native=false

as a jvm property.

**Comment by ozborn** [comment]
Did you try running the last line of test_bug.bsh with force_native set to false? The error still goes through on that. 
I set up the natives, but must have mis-configured. I assume 
```o.s.v.DataBinder - DataBinder requires binding of required fields [rootFile]```
means it can't find the native libraries?

**Comment by agibsonccc** [comment]
How are you running it?

**Comment by ozborn** [comment]
java -cp ./:target/classes:target/deeplearning4j-examples-0.4-rc0-SNAPSHOT.jar org.deeplearning4j.examples.word2vec.Word2VecRawTextExample5

The 3rd line in test_bug.bsh.

**Comment by agibsonccc** [comment]
@ozborn Just wait till the next RC then. This isn't related to anything that's actually a bug. It's just version clashes or native problems (You're welcome to get the native version up and running if you'd like) I'm going to close this.
